planner/core/logical_plan_builder.go

```diff
@@ -6826,6 +6827,15 @@ func (b *PlanBuilder) buildWindowFunctionFrameBound(_ context.Context, spec *ast
 	if err != nil {
 		return nil, err
 	}
+	if bound.CalcFuncs[0].GetType() != col.GetType() {
```

Comment:
I think you can use EvalType to compare here because in my understanding, if `EvalType` is the same, TiDB can compare directly.

Reply:
> I think you can use EvalType to compare here because in my understanding, if `EvalType` is the same, TiDB can compare directly.

okk

Reply:
> I think you can use EvalType to compare here because in my understanding, if `EvalType` is the same, TiDB can compare directly.

okk